### ISSUE TITLE
Compiling setup.py in a nested folder produces setup.txt

### ISSUE THREAD

**Issue by atugushev** [jazzband/pip-tools#1312]
#### Environment Versions

1. OS Type: macOS
1. Python version: `Python 3.8.0`
1. pip version:  `pip 20.2.2`
1. pip-tools version: `pip-compile, version 5.5.0`

#### Steps to replicate

1. Create a folder:
```console
$ mkdir -p foo/bar
```
2. Create `setup.py`:
```console
$ cat << EOF > foo/bar/setup.py
from setuptools import setup
setup(name="bar", version="0.1", install_requires=["six"])
EOF
```
3. Run `pip-compile`:
```console
$ pip-compile foo/bar/setup.py
#
# This file is autogenerated by pip-compile
# To update, run:
#
#    pip-compile foo/bar/setup.py
#
six==1.15.0
    # via bar (foo/bar/setup.py)
```

#### Expected result

```console
$ ls foo/bar
setup.py  requirements.txt
```

#### Actual result

```console
$ ls foo/bar
setup.py  setup.txt
```

**Comment by peymanslh** [comment]
Hi @atugushev 
Can I work on this issue?

**Comment by atugushev** [comment]
@peymanslh sure! Feel free to.

**Comment by AndydeCleyre** [comment]
This also happens to be fixed by the current state of #1329, though the annotations specify `(setup.py)` rather than `(foo/bar/setup.py)`, since the files are adjacent.
